### PR TITLE
Aryan contri3

### LINKED ISSUE #23044: [Bug]: Inconsistent `axis` Normalization in `ivy.gather` Across Backends
### Bug Explanation

The current implementation of the `gather` function in Ivy normalizes the axis parameter differently than native implementations in frameworks like `TensorFlow` , `PyTorch` or `Paddle`. Specifically, the line `axis = axis % len(params.shape)` may not be in line with how` TensorFlow` , `PyTorch` or `Paddle` handle `axis` normalization.

### Steps to Reproduce Bug

we're setting the value of `axis` to be: `axis = axis % len(params.shape)` across all backend at `ivy`
For `jax` backend:   https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/jax/general.py#L123

For `torch` backend: https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/torch/general.py#L174
For `tensorflow` backend: https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/tensorflow/general.py#L103
For `NumPy` backend:https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/numpy/general.py#L75
For `Paddle` backend:https://github.com/unifyai/ivy/blob/16d690d318bb0044378be534cf621b2c4bb9160d/ivy/functional/backends/paddle/general.py#L149

Now, this is how  `axis` value is being set at various framework native APIs:

for `torch`:
At `Pytorch` [gather](https://github.com/pytorch/pytorch/blob/4146be192ead477360a2763c5005e46a9485c3bf/torch/_inductor/lowering.py#L2472) we use a helper called : [_validate_dim](https://github.com/pytorch/pytorch/blob/4146be192ead477360a2763c5005e46a9485c3bf/torch/_inductor/lowering.py#L1100C4-L1100C18)

for `TensorFlow`:
for [tf.gather](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/ragged/ragged_gather_ops.py#L99) we use a helper called [get_positive_axis](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L6659)

For [paddle.gather](https://github.com/PaddlePaddle/Paddle/blob/release/2.5/python/paddle/tensor/manipulation.py#L2721) we set `axis = 0` 

### Environment

Linux + Docker + vscode

### Ivy Version

1.1.9

### Backend

- [X] NumPy
- [X] TensorFlow
- [X] PyTorch
- [X] JAX

### Device

CPU

### PR DESCRIPTION
**Pull request by Aryan8912**
# PR Description

## Related Issue

Close #23044 

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/data_classes/array/experimental/manipulation.py [hunk 1]
```diff
@@ -173,6 +173,11 @@ def vstack(
             ivy.array([[1, 2],
                        [5, 6],
                        [7, 8]])
+
+                    x = ivy.array([1, 2])
+                    y = [ivy.array([[5, 6]]), ivy.array([[7, 8]])]
+                    ivy.vstack((x, y))
+                    ivy.vstack((x, y, x, y))
         """
```
Comment (AnnaTz) [inline ivy/data_classes/array/experimental/manipulation.py:181]: Hi @Aryan8912, I think this docstring edit and the edit in the `rst` file are not related to the PR. We generally want to keep each PR focused on a specific issue. Could you remove those?

File: ivy/functional/backends/jax/general.py [hunk 2]
```diff
@@ -129,7 +152,7 @@ def gather(
     batch_dims: int = 0,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    axis = axis % len(params.shape)
+    axis = get_positive_axis_for_gather(axis, params.ndim)
     batch_dims = batch_dims % len(params.shape)
```
Comment (AnnaTz) [inline ivy/functional/backends/jax/general.py:156]: As was discussed on #26624, you will need to make the same change to all backends and also change the default value of the `axis` argument to 0.

File: ivy/functional/backends/jax/general.py [hunk 3]
```diff
@@ -129,7 +152,7 @@ def gather(
...
     ivy.utils.assertions.check_gather_input_valid(params, indices, axis, batch_dims)
```
Comment (AnnaTz) [inline ivy/functional/backends/jax/general.py:157]: Also, please make sure to search for all the places that `ivy.gather` is being used in our codespace and change the `axis` value as needed. 
What I mean is, if `ivy.gather` is being called using the current default `axis` value, you will need to set it to `axis=-1` to make up for changing the default. I hope this makes sense!

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/backends/jax/general.py (hunk 3 above)
Comment (Aryan8912) [reply ivy/functional/backends/jax/general.py:157]: ok will fix it give 2 days

**Comment by Ishticode** [comment]
Closing this due to inactivity for over 6 months after changes were requested. Please feel free to reopen if you would like to continue working on this. Thank you :)
